### PR TITLE
Add option to notify on orquesta task completion

### PR DESCRIPTION
**Pull request by m4dcoder**
Add a runner option named notify to allow user to specify notification on task completion.

### CODE REVIEW

File: contrib/runners/orquesta_runner/tests/unit/test_notify.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    'publish_state',
+    mock.MagicMock(side_effect=mock_lv_ac_xport.MockLiveActionPublisher.publish_state))
+@mock.patch.object(
+    wf_ex_xport.WorkflowExecutionPublisher,
+    'publish_create',
+    mock.MagicMock(side_effect=mock_wf_ex_xport.MockWorkflowExecutionPublisher.publish_create))
+@mock.patch.object(
+    wf_ex_xport.WorkflowExecutionPublisher,
+    'publish_state',
+    mock.MagicMock(side_effect=mock_wf_ex_xport.MockWorkflowExecutionPublisher.publish_state))
```
Comment (Kami) [inline contrib/runners/orquesta_runner/tests/unit/test_notify.py]: That's a lot of mocking.

Are all those needed and there is no way around that? (in the past while fixing some test issues I noticed a lot of tests have copied mocks from other places and a lot were actually not needed.

File: contrib/runners/orquesta_runner/orquesta_runner/runner.yaml [hunk 2]
```diff
@@ -4,6 +4,11 @@
   enabled: true
   name: orquesta
   runner_module: orquesta_runner
+  runner_parameters:
+    notify:
+      type: array
```
Comment (Kami) [inline contrib/runners/orquesta_runner/orquesta_runner/runner.yaml]: Probably also not a bad idea to specify type for each array item aka ``items: type: string``.

File: contrib/runners/orquesta_runner/orquesta_runner/runner.yaml [hunk 3]
```diff
@@ -4,6 +4,11 @@
   enabled: true
   name: orquesta
   runner_module: orquesta_runner
+  runner_parameters:
+    notify:
```
Comment (Kami) [inline contrib/runners/orquesta_runner/orquesta_runner/runner.yaml]: For action chain and mistral we also support ``skip_notify`` runner parameter, right?

It seems like a good idea to also support that here for consistency reasons, or?

Comment (m4dcoder) [reply contrib/runners/orquesta_runner/orquesta_runner/runner.yaml]: Nope, it's not a good idea. The skip notify and notify conflicts. `skip_notify` assumes notification is default on all task completion.  `notify` is the opposite.  @LindsayHill agree with me here that we should not opt-in folks into task notification.  it's a real hassle to have to explicitly turn off notification for tasks.  it's more common to don't want notification.

File: contrib/runners/orquesta_runner/tests/unit/test_notify.py (hunk 1 above)
Comment (m4dcoder) [reply contrib/runners/orquesta_runner/tests/unit/test_notify.py]: @Kami it is the right set of mocking. You should take a closer look at how I write unit tests for orquesta.  i'm only mocking MQ/messaging here. I'm trying NOT to mock any actual functionality. Once a live action is requested, the unit tests will actually run the workflow and the actions to completion and test rather the notification is called or not.

File: contrib/runners/orquesta_runner/orquesta_runner/runner.yaml (hunk 3 above)
Comment (Kami) [reply contrib/runners/orquesta_runner/orquesta_runner/runner.yaml]: I agree about that behavior as well - I was never a fan of opt-out, should have been opt-in.

Having said that, we just need to be explicit about that "behavior change" in the docs / upgrade notes (well, it's not really a behavior change, but inconsistency between 3 runners).

Comment (Kami) [reply contrib/runners/orquesta_runner/orquesta_runner/runner.yaml]: Let's please document that behavior somewhere, besides that, LGTM :+1:

Comment (m4dcoder) [reply contrib/runners/orquesta_runner/orquesta_runner/runner.yaml]: Yep. I will document this.

**Pull request merged**